File: python/pycommon/infrastructures/rmq_request.py
from pycommon.interfaces.IRequest import IRequest
import pika
import json
import jsonpickle


class RMQRequest(IRequest):
    def __init__(self, rmq_connection, rmq_channel, matched_route, correlation_id, body, reply_to):
        self.rmq_connection = rmq_connection
        self.rmq_channel = rmq_channel
        self.payload_body = None
        self.matched_route = matched_route
        self.correlation_id = correlation_id
        self.body = body
        self.reply_to = reply_to

    def get_body(self):
        return self.body

    def reply_back(self, data):
        self.rmq_channel.basic_publish(exchange="", routing_key=str(self.reply_to),
                                       properties=pika.BasicProperties(correlation_id=self.correlation_id),
                                       body=data)

File: go/common/infrastructures/RMQRequest.go
package infrastructures

import (
	"encoding/json"
	"github.com/streadway/amqp"
	"sample_rmq/common/interfaces"
)

type RMQRequest struct {
	RMQCh         *amqp.Channel
	RMQConn       *amqp.Connection
	Body          map[string]interface{}
	CorrelationID string
	MatchedRoute  string
	ReplyTo       string
	RequestEnded  bool
}

func (request *RMQRequest) GetID() string {
	return ""
}

func (request *RMQRequest) IsEnded() bool {
	return request.RequestEnded
}

func (request *RMQRequest) Next() {
}

func (request *RMQRequest) GetBody() map[string]interface{} {
	return request.Body
}

// ReplyBack responds back to the request
func (request *RMQRequest) ReplyBack(responseCode int, responseBody interfaces.RouterResponse) bool {
	bytes, _ := json.Marshal(responseBody)
	request.RequestEnded = true
	_ = request.RMQCh.Publish(
		"",              // Exchange global
		request.ReplyTo, // routing key
		false,           // mandatory
		false,           // immediate
		amqp.Publishing{
			CorrelationId: request.CorrelationID,
			Body:          bytes,
		})
	return true
}

func (request *RMQRequest) GetHeaders(key string) interface{} {
	return nil
}

func (request *RMQRequest) GetEndpoint() string {
	return ""
}

func (request *RMQRequest) SetContextData(key string, data interface{}) {
}

func (request *RMQRequest) GetContextData(key string) interface{} {
	return nil
}

func (request *RMQRequest) GetParam(key string) interface{} {
	return nil
}

func (request *RMQRequest) GetFile(key string) (interface{}, interface{}) {
	return nil, nil
}
